lib/internal/bootstrap/node.js

```diff
@@ -651,6 +655,26 @@
     }
   }
 
+  // When a path is provided to the CLI the trailing slash must be preserved.
+  // Otherwise cjs can load an incorrect file.
+  function hasTrailingSlash(filename, path) {
+    let trailingSlash = false;
+    if (filename && filename.length) {
+      // we do not want to add a slash to a root path `/`
+      // this is the only path where path.resolve returns a slash at the end
+      trailingSlash = filename.length > 1 &&
+        filename.charAt(filename.length - 1) === path.sep;
+      if (!trailingSlash) {
+        if (path.sep === '/') {
+          trailingSlash = /(?:^|\/+)\.?\.$/.test(filename);
+        } else {
+          trailingSlash = /(?:^|\\+)\.?\.$/.test(filename);
```

Comment:
Can you clarify what these cases are for?

Reply:
Paths can be passed to the cli with windows or *nix separators but can't be mixed.
I chose to copy the regex literal replacing only the slash instead of compiling a new RegExp because i felt it was more readable.

think i should add a comment above each like this
`//support windows paths`
so it's a bit more obvious what i intend?

i cargo culted the `?:` not capturing group from the regex in internal/modules/cjs/loader.js. I'm not sure why it's there and am rather curious why @bmeck included it in the original.

Reply:
I needed a group to alternate `^` and a separator to avoid matching things like `"a/b.."`, made non-capture because I don't care about which it is.

Reply:
Ok great. works as intended

Reply:
Lines 668-674 could be replaced with:

```js
// Separator is either \ or /. However, user can monkey-patch other values. So
// make sure it's either \ or /, but nothing else.
const separator = path.sep === '\\' ? '\\' : '/';
trailingSlash = new RegExp(`(?:^|\\${separator}+)\\.?\\.$`)).test(filename);
```